### DISCUSSION TITLE
Bagaimana caranya concat icon dengan text pada ui.R ?

### DISCUSSION THREAD

**Discussion by vidric** [teamalgoritma/community#571] [Q&A] [answered]
Sy membuat checkbox pada ui.R dengan code sebagai berikut :
`box(
            width = 4,
            checkboxGroupInput(
              inputId = "select_input",
              label = "Choose League",
              # choices = c("0", "1", "2", "3", "4"),
              selected = c("0", "1", "2", "3", "4"),
              #inline = FALSE,
              #width = NULL,
              choiceNames = list(icon("square",class = "fa-solid", style = "color: #CD7F32;"), "Silver", "Gold", "Diamond", "Champion"),
              choiceValues = list("0", "1", "2", "3", "4")
            )
          )`

Sy ingin menampilkan icon dengan text pada setiap label checkbox. Bagaimana caranya ya?

**Comment by idoali** [comment, marked as answer]
Terima kasih pak atas pertanyaannya. 

Ada beberapa hal yang perlu dicatat pak : 
- Ketika menggunakan parameter `choices =`, kita tidak perlu lagi menggunakan `choiceNames` dan `choiceValues` karena `choices` adalah gabungan keduanya.
- `selected =` adalah pilihan default pak, jadi cuma bisa memiliki satu nilai 
- Untuk menggabungkan text dengan icon, kita bisa memanfaatkan tags dari HTML. Di kasus ini saya akan menggunakan `p()`.

Dengan memperbaiki code bapak berdasarkan poin-poin diatas, maka code bapak akan menjadi seperti dibawah : 

```{R}
checkboxGroupInput(inputId = "select_input",
                   label = "Choose League",
                   selected = "0",
                   inline = FALSE,
                   choiceNames = list(p(icon("square",class = "fa-solid", style = "color: #CD7F32;"), "Bronze"),
                                      p(icon("square",class = "fa-solid", style = "color: #CD7F32;"), "Silver"),
                                      p(icon("square",class = "fa-solid", style = "color: #CD7F32;"), "Gold"),
                                      p(icon("square",class = "fa-solid", style = "color: #CD7F32;"), "Diamond"),
                                      p(icon("square",class = "fa-solid", style = "color: #CD7F32;"), "Champion")),
                   choiceValues = list("0", "1", "2", "3", "4") )
```

*note : Pada code diatas, semua pilihan masih memiliki warna yang sama*
